numpy/core/src/private/npy_sort.h

```diff
@@ -190,5 +190,8 @@ int amergesort_timedelta(npy_timedelta *vec, npy_intp *ind, npy_intp cnt, void *
 int npy_quicksort(void *base, size_t num, size_t size, npy_comparator cmp);
 int npy_heapsort(void *base, size_t num, size_t size, npy_comparator cmp);
 int npy_mergesort(void *base, size_t num, size_t size, npy_comparator cmp);
```

Comment:
Might be worth using the same signature for both. I think we are safe for structured types, etc, as the arrays are created with new comparators.